### DISCUSSION TITLE
Custom Sign up page (like Vercel's sign up page) - my current sign up page is super slow

### DISCUSSION THREAD

**Discussion by dubbajubba** [nextauthjs/next-auth#4232] [Help]
Hi there,

I'm trying to build a custom sign up page, similar to Vercel's sign up page. https://vercel.com/signup
<img width="1023" alt="image" src="https://user-images.githubusercontent.com/33437206/159405506-6cb5cf5b-ca34-4539-97d2-d9f07fd12439.png">

Before I continue, I first want to mention that I have seen these two discussions, but I don't know if they are similar enough to my question and that is why I am starting a new discussion.
https://github.com/nextauthjs/next-auth/discussions/4124
https://github.com/nextauthjs/next-auth/discussions/945

Maybe I am missing something, but I am not sure why there isn't something similar to this already built into next-auth. I know that next-auth automatically creates an account for a user on initial sign in, which I love, but users naturally look for a sign up page. From a user experience perspective, this is very common and expected.

### Problem
I have a sign up page that looks great, and handles the sign up flow. However, the page is extremely slow to load when it is first viewed. It loads much faster after the first load (I'm assuming this has something to do with the way Vercel creates its caches).  

I believe the slow behavior is being caused by the `getProviders()` function I am using in `getServerSideProps()`. In contrast, my sign in page loads fine.

I don't know exactly what is causing the `getProviders()` function to be so slow, but I think that it has something to do with how my sign up page is set up. I have a feeling it is establishing a separate DB connection from the next-auth api functions. I am using Azure's CosmosDB for my database, hosted on Azure.

If anyone has any thoughts, or suggestions on improving my code, it is much appreciated.

The anchor tag that loads the page:
```TSX
<Link href="/signup">
  <a className='btn btn--type-primary'>Sign up</a>
</Link>
```

**/pages/signup.tsx**
```TSX
import { useEffect } from 'react';
import { GetServerSideProps } from 'next';
import { getProviders, signIn, useSession } from 'next-auth/react';
import { getCsrfToken } from 'next-auth/react';

export default function SignIn({
  providers,
  csrfToken,
}: {
  providers: any[];
  csrfToken: any[];
}) {
  const { data: session, status } = useSession();
  const loading = status === 'loading';
  
  useEffect(()=>{
    if(!loading){
      if (session) {
        router.push('/qbits')
      }
    }
  },[session,loading]);

  if(!loading && !session)
  {
    return (
      <>
        {/* CODE REMOVED FOR BREVITY. CAN PROVIDE IF NECESSARY */}
      </>
    );
  }

  return (
    <div></div>
  )
}

//This is the recommended way for Next.js 9.3 or newer
export const getServerSideProps: GetServerSideProps = async context => {
  const providers = await getProviders();
  const csrfToken = await getCsrfToken(context);

  return {
    props: { providers, csrfToken },
  };
};
```

Please let me know if I need to provide more details.

**Comment by dubbajubba** [comment]
Still looking for answers on this 😄 

Any help or suggestions are very appreciated! 👍

**Reply by prichodko** [reply]
I don't see any reason to fetch Providers for every user that visits your page (e.g., they don't change during the lifetime of your application). So instead, you can move it to the `getStaticProps` or hard-code them into your code.

**Reply by dubbajubba** [reply]
Thanks for your response. I also thought that wasn't necessary, but I left it in because that is what the documentation suggests.
https://next-auth.js.org/configuration/pages#oauth-sign-in
